core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1150,6 +1155,7 @@ class ReplicaManager(val config: KafkaConfig,
               if (partition.futureReplicaDirChanged(destinationDir)) {
                 replicaAlterLogDirsManager.removeFetcherForPartitions(Set(topicPartition))
                 partition.removeFutureLocalReplica()
+                logManager.resumeCleaning(topicPartition)
```

Comment:
In theory, it's possible that immediately after the `partition.futureReplicaDirChanged` check, the future partition catches up and is replaced as the current partition. During that process, `logManager.resumeCleaning()` has already been called. Calling it a second time here will lead to `IllegalStateException`. We could potentially check for the future replica again after `replicaAlterLogDirsManager.removeFetcherForPartitions()`. At that point, if the future replica still exist, it's guaranteed to be there afterward.